public/login.html

```diff
@@ -80,3 +81,18 @@ <h1>Login</h1>
 
   <button type="submit">Login</button>
 </form>
+
+<script>
+  function login(e) {
+    e.target.action = 'https://api.dev.core.ac.uk/login_check?continue=' + encodeURIComponent(
+      window.location.origin
+    );
+  }
+
+  const urlParams = new URLSearchParams(window.location.search);
+  if (urlParams.has('reason')) {
+    const message = document.getElementById('error-message')
+    message.innerHTML='The username or password you entered is incorrect.';
+    message.style.display = 'block';
```

Comment:
Simple CSS snippet may help you avoid this.

```css
.message:empty { display: none }
```

Reply:
This is really cool selector. Thanks